packages/core/src/dialects/postgres/query-generator-typescript.ts

```diff
@@ -247,4 +258,30 @@ export class PostgresQueryGeneratorTypeScript extends AbstractQueryGenerator {
 
     return fragment;
   }
+
+  deleteQuery(tableName: TableNameOrModel, options: DeleteQueryOptions) {
```

Comment:
snowflake, postgres and mssql have almost identical implementations. MySQL and MariaDB as well

I think the code should be deduplicated by either:
- moving the code to `AbstractQueryGeneratorInternal` (same concept as the QI one) with a parameter to indicate whether the dialect supports `LIMIT` (could be a support flag)
- or having it in `AbstractQueryGenerator` and changing the query based on support flags

mssql would still need to override, but could call the method / `super` and append `; SELECT @@ROWCOUNT AS AFFECTEDROWS;` to the result

Reply:
Does SQLite still need an override (it uses the rowid for the limit query) or shall I merge to to function similar to postgres and snowflake?

Reply:
I'd say SQLite should have an override as its logic is substantially different from the others